bots/code_review_master.rb

```diff
@@ -0,0 +1,202 @@
+  LIST_COMMAND = 'list'
+
+  def add_callbacks(slack_connection)
+    slack_connection.on_message do |user:, channel:, message:, timestamp:, thread_ts:|
+      if codereviewmaster?(user, message, channel, timestamp, thread_ts)
+        data = CodeReviewMaster::Data.new(channel, user)
+        m = CodeReviewMaster::Message.new(
+          slack_connection: slack_connection,
+          channel: channel,
+          thread_ts: thread_ts
+        )
+
+        ################
+        # Add reviewer #
+        ################
+        if (message_segment = add_reviewer_request?(message))
+          SlackbotFrd::Log.debug(
+            "user '#{user}' in channel '#{channel}' is adding reviewer with string #{message_segment}"
+          )
+          if (added_user = data.add_reviewer!(message_segment))
+            m.send_message("Added #{added_user} to the list of reviewers :shipit:")
+          else
+            m.send_message("Can't add a user with string #{message_segment} :fail:")
+          end
+        end
+
+        ###################
+        # Remove reviewer #
+        ###################
+        if (to_remove = remove_reviewer_request?(message))
+          SlackbotFrd::Log.debug(
+            "user '#{user}' in channel '#{channel}' is adding the reviewer #{to_remove}"
+          )
+          if data.remove_reviewer!(to_remove)
+            m.send_message("Removed #{to_remove} from the list of reviewers :shipit:")
+          else
+            m.send_message("Can't remove #{to_remove} from the list of reviewers :fail:")
+          end
+        end
+
+        ##################
+        # List reviewers #
+        ##################
+        if list_reviewers_request?(message)
+          SlackbotFrd::Log.debug(
+            "user '#{user}' in channel '#{channel}' is listing reviewers"
+          )
+          if data.reviewers.any?
+            m.send_message("Reviewers are: #{data.users.join(', ')}")
+          else
+            m.send_message("No reviewers yet! Add one like `codereviewmaster add <name>`")
+          end
+        end
+
+        ###################
+        # Clear reviewers #
+        ###################
+        if clear_reviewers_request?(message)
+          SlackbotFrd::Log.debug(
+            "user '#{user}' in channel '#{channel}' is clearing all reviewers"
+          )
+          if data.reviewers.any? && data.clear_reviewers!
+            m.send_message("Reviewers have been cleared!")
+          else
+            m.send_message("No reviewers yet! Add one like `codereviewmaster add <name>`")
+          end
+        end
+
+        ####################
+        # Output help text #
+        ####################
+        if help_output_request?(message)
+          SlackbotFrd::Log.debug(
+            "user '#{user}' in channel '#{channel}' is requesting help output"
+          )
+          help = <<-HELP_TEXT
+Hi! I help assign code reviewers.
+
+*Assign a reviewer*
+
+`codereviewmaster g/123456`
```

Comment:
Nitpick and not required for merge: It's a little confusing as to who's being added as a reviewer. Might want to talk about how it adds a group of users that you specifically define as documented below.